### PR TITLE
UPSTREAM: <carry>: openshift: Rework TestControllerFindMachineByID

### PR DESCRIPTION
**Pull request by frobware**
Simplify test setup using `newMachineController()`.

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go [hunk 1]
```diff
@@ -107,64 +106,64 @@ func makeMachineOwner(i int, replicaCount int, annotations map[string]string, ow
...
-	// Verify machine count goes to 1.
-	machines, err = controller.machineInformer.Lister().Machines(v1.NamespaceAll).List(labels.Everything())
-	if err != nil {
-		t.Fatalf("unexpected error, got %v", err)
-	}
-	if len(machines) != 1 {
-		t.Fatalf("expected 1 machine, got %d", len(machines))
+	type testCase struct {
+		description    string
+		name           string
```
Comment (ingvagabund) [inline cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go]: Nit: add comment the `name` is used as a namespace for `newMachineSetTestObjs`. Thus, it needs to compliant with namespace naming.

Comment (frobware) [reply cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go]: `tc.name` is not passed to `newMachineSetTestObjs()`. What is passed is `t.Name()` which is the name of this function under Test. And that, as you point out is not compliant as it uses upper case characters. Given that we have just merged a number of PRs doing exactly the same thing can I follow up and fix all these in one go. One solution would be to do `lowercase(t.Name())`.

Comment (frobware) [reply cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go]: > Nit: add comment the `name` is used as a namespace for `newMachineSetTestObjs`. Thus, it needs to compliant with namespace naming.

I misread this. for `name` you meant `t.Name()`. See - https://github.com/openshift/kubernetes-autoscaler/pull/61/files#r266810534

**Pull request merged**